apps/emqx_management/test/emqx_mgmt_api_clients_SUITE.erl

```diff
@@ -171,6 +215,290 @@ t_clients(_) ->
+            %% 2) Client connects to *another node* and takes over, listed only once.
+            ?tp(notice, "scenario 3", #{}),
+            O = #{api_port => APIPort},
+            ClientId = <<"c3">>,
+            C1 = connect_client(#{port => Port1, clientid => ClientId}),
+            assert_single_client(O#{node => N1, clientid => ClientId, status => connected}),
+            unlink(C1),
+            C2 = connect_client(#{port => Port2, clientid => ClientId}),
+            assert_single_client(O#{node => N2, clientid => ClientId, status => connected}),
+            %% Doesn't show up in the other node while alive
+            ?retry(
+                100,
+                20,
+                ?assertMatch(
+                    {ok, {{_, 200, _}, _, #{<<"data">> := []}}},
+                    list_request(APIPort, "node=" ++ atom_to_list(N1))
+                )
+            ),
+            ok = emqtt:stop(C2),
+            ok = erpc:call(N1, emqx_persistent_session_ds, destroy_session, [ClientId]),
+
+            ok
+        end,
+        []
+    ),
+    ok.
+
+t_persistent_sessions4(Config) ->
+    [N1, N2] = ?config(nodes, Config),
+    APIPort = 18084,
+    Port1 = get_mqtt_port(N1, tcp),
+    Port2 = get_mqtt_port(N2, tcp),
+
+    ?assertMatch({ok, {{_, 200, _}, _, #{<<"data">> := []}}}, list_request(APIPort)),
+
+    ?check_trace(
+        begin
+            %% Scenario 4
+            %% 1) Client connects and is listed as connected.
+            %% 2) Client disconnects and is listed as disconnected.
+            %% 3) Client reconnects to *another node* and is listed as connected once.
+            ?tp(notice, "scenario 4", #{}),
+            O = #{api_port => APIPort},
+            ClientId = <<"c4">>,
+            C1 = connect_client(#{port => Port1, clientid => ClientId}),
+            assert_single_client(O#{node => N1, clientid => ClientId, status => connected}),
+            ok = emqtt:stop(C1),
+            %% While disconnected, shows up in both nodes.
+            assert_single_client(O#{node => N1, clientid => ClientId, status => disconnected}),
+            assert_single_client(O#{node => N2, clientid => ClientId, status => disconnected}),
+            C2 = connect_client(#{port => Port2, clientid => ClientId}),
+            assert_single_client(O#{node => N2, clientid => ClientId, status => connected}),
+            %% Doesn't show up in the other node while alive
+            ?retry(
+                100,
+                20,
+                ?assertMatch(
+                    {ok, {{_, 200, _}, _, #{<<"data">> := []}}},
+                    list_request(APIPort, "node=" ++ atom_to_list(N1))
+                )
+            ),
+            ok = emqtt:stop(C2),
+            ok = erpc:call(N1, emqx_persistent_session_ds, destroy_session, [ClientId]),
+
+            ok
+        end,
+        []
+    ),
+    ok.
+
+t_persistent_sessions5(Config) ->
+    [N1, N2] = ?config(nodes, Config),
+    APIPort = 18084,
```

Comment:
Why is it using non-standard port?

Reply:
Because the suite starts the management on the default port for the other tests.

Reply:
And why can't we reuse the default setup? 
Also, is it possible to call the Erlang interface directly, bypassing from/to JSON conversion?

Reply:
> And why can't we reuse the default setup?

Those tests are using an isolated, clustered setup.  To avoid having to stop the running applications from the suite, it's simpler to just set another port here.

> Also, is it possible to call the Erlang interface directly, bypassing conversion from/to JSON and the HTTP?

I believe it's best to test the user-facing API in this case, since it's the object of the test.  Specially during the bridge refactoring tasks, we observed that tests that don't use such APIs tend to hinder future changes to the code.